### PR TITLE
Add linter support for named anchors

### PR DESCRIPTION
**Pull request by andyfmiller**
Linter local-refs-exist rule does not recognize named anchors. This is flagged as a missing reference:
```
<a name="API"></a> API
<a href="#API">See API</a>
```

PR adds support for named anchors to local-refs-exist.js and test to local-refs-exist-spec.js.

### CODE REVIEW

**Comment by sidvishnoi** [comment]
Hi @andyfmiller 
Please run `npm run lint -- --fix` to fix those linting errors :slightly_smiling_face: 

Curious to know how how named anchors are being used. Can you link a document?

File: tests/spec/core/linter-rules/local-refs-exist-spec.js [hunk 1]
```diff
@@ -51,4 +51,21 @@ describe("Core Linter Rule - 'local-refs-exist'", () => {
     const result = await rule.lint(config, doc);
     expect(result).toBeUndefined();
   });
+
+  it("handles named anchors", async () => {
+    doc.body.innerHTML = `
+      <a name="NAME">PASS</a>
+      <a href="#NAME">PASS</a>
+      <a href="#NAME-NOT-EXIST">FAIL</a>
+      <a href="#NAME-NOT-EXIST">FAIL</a>
```
Comment (sidvishnoi) [inline tests/spec/core/linter-rules/local-refs-exist-spec.js]:
```suggestion
```
Can remove this.

Comment (andyfmiller) [reply tests/spec/core/linter-rules/local-refs-exist-spec.js]: > Curious to know how how named anchors are being used. Can you link a document?

Typically in markdown documents. For example, https://raw.githubusercontent.com/IMSGlobal/caliper-spec/develop/guides/certification-guide.md

**Comment by andyfmiller** [comment]
> Curious to know how how named anchors are being used. Can you link a document?

Typically in markdown documents. For example, https://raw.githubusercontent.com/IMSGlobal/caliper-spec/develop/guides/certification-guide.md

File: tests/spec/core/linter-rules/local-refs-exist-spec.js (hunk 1 above)
Comment (andyfmiller) [reply tests/spec/core/linter-rules/local-refs-exist-spec.js]: Removed and removed check for 2 occurrences (occurrences already checked by first test in this rule).

**Comment by sidvishnoi** [comment]
Interesting. This is not however we were expecting ReSpec being used :sweat_smile:

If you're interested in referencing the headings with a unique custom id, you can use:
``` diff
- ## <a name="contributors"></a>Contributors
+ ## Contributors {#contributors}
```

For references, if you use the following syntax, they'll be automatically linked and get listed at the end of page:
``` diff
- [RFC 2119](#rfc2119)
+ [[RFC2119]]
```

**Comment by sidvishnoi** [comment]
Though, happy to have the flexibility.

**Comment by andyfmiller** [comment]
Those are great tips! I'll pass those on to the author.

**Comment by sidvishnoi** [comment]
@andyfmiller Let me know if this feature is still needed or the above fixed your issues.

**Comment by amiller-ims** [comment]
> @andyfmiller Let me know if this feature is still needed or the above fixed your issues.

While the syntax `{#custom-id}` works, we have very many documents that use the named anchor format. It would be convenient if we could transition over time. If respec supported both formats, that would be ideal.

**Comment by saschanaz** [comment]
>we have very many documents that use the named anchor format.

Could you provide some example? We are happy to support existing documents but first we need to know how many documents use it, to minimize our maintenance burden.

**Comment by andyfmiller** [comment]
The authors/editors here that are affected feel they can live with the linter warning until they change to the recommended format.

**Comment by saschanaz** [comment]
Okay, please ping with the specific real world examples if you feel you need this.

**Comment by marcoscaceres** [comment]
Although doing `<a name>` is super ol' school, this is something we might also do for adding "old ids" (I've also seen this done in various specs)... so it wouldn't be a bad linter to have.

**Pull request merged**